staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -1361,7 +1361,7 @@ func buildOpenAPIModelsForApply(staticOpenAPISpec *spec.Swagger, crd *apiextensi
 	specs := []*spec.Swagger{}
 	for _, v := range crd.Spec.Versions {
 		// Defaults are not pruned here, but before being served.
-		s, err := builder.BuildSwagger(crd, v.Name, builder.Options{V2: false, StripValueValidation: true, StripNullable: true, AllowNonStructural: false})
+		s, err := builder.BuildSwagger(crd, v.Name, builder.Options{V2: true, OverridePruneV2Fields: true, StripValueValidation: true, StripNullable: true, AllowNonStructural: false})
```

Comment:
why those settings? This needs a go comment

Reply:
https://github.com/kubernetes/kubernetes/pull/105945#discussion_r746235701